### ISSUE TITLE
Upgrade Password Hashing

### ISSUE THREAD

**Issue by TBSliver** [Pear-Trading/Registration#17]
Currently password hashing uses 'salted' md5 hashes. This needs to be upgraded to use Blowfish encryption.

The server has been tested for compatibility with Blowfish Crypt, however the php version does not have password_hash so will need that to be implemented as a function for the crypt function, with random salts.

This will also affect the mobile application, so will have to be done at the same time - and preferably before actually allowing users onto the application.

This will not take a massive amount of time, but will obviously need testing.

**Comment by MichaelHallam** [comment]
Let me know if I have to do anything regarding this Tom.
https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail
This
email has been sent from a virus-free computer protected by Avast.
www.avast.com
https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail
<#DDB4FAA8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by TBSliver** [comment]
Have upgraded the app to run under FastCGI - this allowed us to upgrade from 5.3.3 php to 5.5.\* (unsure which to be honest but meh), which gives us the password_hash ability, saving having to write a compatibility layer. As a result, I have removed all the old crappy md5 hashing of passwords from the app, and we are now using Blowfish cryptography. This has also been updated for the android app and has been tested as working! done!
